infrastructure/src/main/java/org/corfudb/infrastructure/ServerHandshakeHandler.java

```diff
@@ -76,7 +76,9 @@ public void channelRead(ChannelHandlerContext ctx, Object m) throws Exception {
             handshake = (CorfuPayloadMsg<HandshakeMsg>) m;
             log.debug("channelRead: Handshake Message received. Removing {} from pipeline.",
                     READ_TIMEOUT_HANDLER);
-            ctx.pipeline().remove(READ_TIMEOUT_HANDLER);
+            // Remove the handler from the pipeline. Also remove the reference of the context from
+            // the handler so that it does not disconnect the channel.
+            ctx.pipeline().remove(READ_TIMEOUT_HANDLER).handlerRemoved(ctx);;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)